hamlpy/hamlpy_watcher.py

```diff
@@ -103,15 +104,28 @@ def watch_folder():
     
     while True:
         try:
-            _watch_folder(input_folder, output_folder, compiler_args)
+            (total_files, num_failed) = _watch_folder(input_folder, output_folder, compiler_args)
+            
+            # Stop after the first round of compiling 
+            if args.once:
```

Comment:
I think that this behavior should be outside of this while loop instead of highjacking it, that will make 2 clear path in the code imo.